int32 Zalver_State
